rag_api/main.py

```diff
@@ -87,12 +92,21 @@ def initialize_vectorstore(doc_loc: str, doc_filter="**/*.*"):
         TXT_SPLITTER = text_splitter.__class__.__name__
 
         all_splits = text_splitter.split_documents(data)
-        logger.debug(f"Split into {len(all_splits)} chunks")
+        logger.info(f"Split into {len(all_splits)} chunks")
 
+        # Initialize the vector store with GPU embedding function 
+        model_kwargs = {'device': 'cuda'}
+        encode_kwargs = {'normalize_embeddings': True}
         vector_store = Chroma.from_documents(
-            documents=all_splits,
-            embedding=HuggingFaceEmbeddings(model_name=settings.EMBEDDING_MODEL),
+            collection_name="Squad2.0",
+            documents=tqdm(all_splits, desc="Vectorizing documents"), #all_splits,
+            embedding=HuggingFaceEmbeddings(
+                model_name=settings.EMBEDDING_MODEL, 
+                model_kwargs=model_kwargs,
+                encode_kwargs=encode_kwargs),
+            persist_directory=doc_loc+"/cache",
```

Comment:
It seems like the documents are reprocessed and vectorized on start-up even if persist_directory is created.  Is it worth keeping the persist_directory if it doesn't add a start-up speed beneift?

Reply:
will add if-cache-exists-skip-vectorize logic next